app/services/calculate_indicators_and_signals.rb

```diff
@@ -0,0 +1,244 @@
+class CalculateIndicatorsAndSignals < Patterns::Service
+  def initialize(coin, limit = 200)
+    @coin = coin
+    @limit = limit
+  end
+
+  def call
+    daily_price_data = parse_daily_prices(@coin)
+    indicators = get_indicator_values(daily_price_data)
+    signals = get_indicator_signals(indicators)
+
+    [indicators, signals]
+  end
+
+  private
+
+  def parse_daily_prices(coin)
+    coin.prices_data
+      .map do |d|
+        { adj_close: d['close'] || 0, high: d['high'] || 0, low: d['low'] || 0 }
+      end
+      .last(@limit) # limit dataset for easier processing
+  end
+
+  def get_indicator_values(data)
+    {
+      rsi: rsi(data).round(0),
+      stochrsi: stochastic_rsi(data).round(0),
+      macd: macd(data).round(0),
+      cci: cci(data).round(0),
+      stochastic_fast: stochastic_fast(data).round(0),
+      stochastic_slow: stochastic_slow(data).round(0),
+      sma: (simple_moving_average(data, 20) - simple_moving_average(data, 50)).round(0),
+      ema: (exponential_moving_average(data, 10) - exponential_moving_average(data, 20)).round(0)
+    }
```

Comment:
Might be easier to just map `round(0)` to each hash element instead of repeating it throughout the hash?